step.yml

```diff
@@ -1,26 +1,57 @@
 ---
-title: Artifact pull
-summary: Step to pull artifacts of a pipeline
+title: Pull Pipeline intermediate files
+summary: Pull intermediate files generated by the Pipeline Builds.
 description: |-
-  The step downloads build artifacts of a pipeline to a local folder.
+  The step downloads intermediate files (build artifacts generated by the pipeline workflows,
+  that are intended to be shared with subsequent workflows of the pipeline) to a local folder.
 
-  By default, all artifacts generated by any workflow of the pipeline are downloaded. This can be limited
-  by setting the `artifact_sources` input variable.
+  These artifacts can be uploaded by the [Deploy to Bitrise.io step](https://www.bitrise.io/integrations/steps/deploy-to-bitrise-io),
+  via specifying the *Files to share between pipeline stages* input.
+
+  Pipeline intermediate files have an environment variable key assigned,
+  after downloading the given artifact, an environment variable (with the assigned key) will point to the file's local path.
+
+  Directories specified in *Files to share between pipeline stages* input will be archived and uploaded as a single file.
+  After downloading, they are extracted, and the specified environment variable will point to the directory's local path.
```

Comment:
Remove this part (was already addressed in other comments before this)

Reply:
The last sentence here is not mentioned earlier, somehow we should keep it.